index.js

```diff
@@ -1,91 +1,55 @@
-  );
-};
-setInterval(setTime, 1000);
-
-class Book {
-  constructor(title, author) {
-    this.title = title;
-    this.author = author;
-  }
+import { DateTime } from './modules/luxon.min.js';
+import Book from './modules/bookclass.js';
+import pageFunctionality from './modules/ui_controller.js';
+
+const getBooks = JSON.parse(localStorage.getItem('books'));
+let bookList = [];
+if (getBooks !== null) {
+  bookList = getBooks;
 }
 
-let Books = [];
-
-window.onload = () => {
-  if (registeredBooks.innerHTML === "") {
-    document.querySelector(".emptymsg").style.display = "block";
-    registeredBooks.style.border = "none";
-  } else {
-    document.querySelector(".emptymsg").style.display = "none";
-  }
-  if (localStorage.getItem("Books")) {
-    Books = JSON.parse(localStorage.getItem("Books"));
-    registeredBooks.style.border = "1px solid black";
-    document.querySelector(".emptymsg").style.display = "none";
-  }
-  addBooks(Books);
-};
+// create a book every time the add book btn is clicked.
+document.addEventListener('submit', (e) => {
+  e.preventDefault();
+  let title = document.getElementById('title').value;
+  let authorInput = document.getElementById('author').value;
 
-//add
+  const newBook = new Book.Book(title, authorInput); // create new book object
+  bookList.push(newBook);
 
-addButton.addEventListener("click", (e) => {
-  if (Title.value === "" || author.value === "") {
-    e.preventDefault();
-  } else {
-    const book = new Book(Title.value, author.value);
-    Books.push(book);
-    addBooks(Books);
-    localStorage.setItem("Books", JSON.stringify(Books));
-    document.querySelector(".emptymsg").style.display = "none";
-    registeredBooks.style.border = "1px solid black";
-  }
+  // send book to local storage
+  newBook.storeBook(bookList);
+  title = '';
+  authorInput = '';
 });
 
-//remove
-
-document.querySelector(".booksdata").addEventListener("click", (e) => {
-  if (!localStorage.getItem("Books")) {
-    document.querySelector(".emptymsg").style.display = "block";
-    registeredBooks.style.border = "none";
-  } else {
-    Books.splice(e.target.id, 1);
-
-    localStorage.setItem("Books", JSON.stringify(Books));
-    deleteBook(e.target);
-  }
+// dynamically showing book in the UI
+const bookShelf = document.querySelector('.book-shelf');
+bookShelf.innerHTML += bookList
+  .map((item) => `
+          <div class="compactment">
+        <ul class="booklist">
+          <li class="title">${item.title}</li>
+          <li class="by">BY</li>
+          <li class="author">${item.author}</li>
+        </ul>
+        <button type="button" class="remove-btn">Remove</button>
+      </div>
+      <hr class="divider" />`)
+  .join('');
+
+const removeBtn = document.querySelectorAll('.remove-btn');
+removeBtn.forEach((item, index) => {
+  item.addEventListener('click', () => {
+    const takeOut = new Book.Book(bookList[index].title, bookList[index].author);
+    takeOut.removeBook(takeOut.title);
+  });
```

Comment:
- [ ] As requested by the previous code reviewer currently, the UI is not getting updated when I add a new book, and also please remember to save the added book in the local storage. Please kindly fix this :pray: